safe_transaction_service/history/tests/test_commands.py

```diff
@@ -258,6 +258,69 @@ def test_reindex_master_copies(self, current_block_number_mock: PropertyMock):
                         from_block_number + block_process_limit * 2,
                     )
                     self.assertEqual(find_relevant_elements_mock.call_count, 2)
+        IndexServiceProvider.del_singleton()
+
+    @mock.patch.object(
+        EthereumClient, "current_block_number", new_callable=PropertyMock
+    )
+    def test_reindex_erc20_events(self, current_block_number_mock: PropertyMock):
+        logger_name = "safe_transaction_service.history.services.index_service"
+        current_block_number_mock.return_value = 1000
+        command = "reindex_erc20"
+
+        with self.assertRaisesMessage(
+            CommandError, "the following arguments are required: --from-block-number"
+        ):
+            call_command(command)
+
+        buf = StringIO()
+        with self.assertLogs(logger_name, level="WARNING") as cm:
+            call_command(
+                command,
+                "--block-process-limit=11",
+                "--from-block-number=76",
+                stdout=buf,
+            )
+            self.assertIn("Setting block-process-limit to 11", buf.getvalue())
+            self.assertIn("Setting from-block-number to 76", buf.getvalue())
+            self.assertIn("No addresses to process", cm.output[0])
+
+        safe_contract = SafeContractFactory()
+        buf = StringIO()
+        with self.assertLogs(logger_name, level="INFO") as cm:
+            with mock.patch.object(
+                Erc20EventsIndexer, "find_relevant_elements", return_value=[]
+            ) as find_relevant_elements_mock:
+                IndexServiceProvider.del_singleton()
+                from_block_number = 100
+                block_process_limit = 500
+                call_command(
+                    command,
+                    f"--block-process-limit={block_process_limit}",
+                    f"--from-block-number={from_block_number}",
+                    stdout=buf,
+                )
+                self.assertIn(
+                    f"Start reindexing addresses {[safe_contract.address]}",
+                    cm.output[0],
+                )
+                self.assertIn("found 0 traces/events", cm.output[1])
+                self.assertIn(
```

Comment:
Maybe include a test with 1 or more events? What do you think?

Reply:
I would like to refactor `IndexService` at some point (to fix the circular imports, between other things) so I don't want to spend much time on it for now 😃